src/publish.js

```diff
@@ -314,7 +314,7 @@ have published it.\
       }
 
 
-      if (currentBranch = repo.getShortHead()) {
+      if (currentBranch == repo.getShortHead()) {
```

Comment:
Suggesting this code change, since I think this is meant to be assignment after all. So, it looks like we can handle it like all the other "no assignment in operand" lints:

```suggestion
      currentBranch = repo.getShortHead();
      if (currentBranch) {
```

---

<details><summary>Long version / breakdown / context (click to expand):</summary>

Looks like this was actually meant to be an assignment rather than a comparison after all. It was assignment back in the CoffeeScript: 

https://github.com/pulsar-edit/ppm/blob/de1221ea743721bb68cf895596528f7a2251a7f9/src/publish.coffee#L252-L253

I think what's happening here is, based on the [`git-utils` documentation](https://github.com/atom/git-utils#readme), `git-utils` generally seems to prefer returning `null` rather than throwing Errors if something weird happens under the hood (?). So you can basically check if something from `git-utils` worked by checking if its result was truthy or null.

So, we could expect the result of `repo.getShortHead()` to be a shortened SHA (string, truthy), or null (falsy). And the `if ()` check is seeing whether the `repo.getShortHead()` succeeded or not/whether we have an actual string value to work with or not.

</details>

I definitely agree with separating out these sorts of things into more lines, so the linter warning about this makes sense.

Reply:
Thanks for doing the research into this one! Fantastic explanation, and seems you are 100% correct